internal/infra/open_telemetry.go

```diff
@@ -0,0 +1,99 @@
+package infra
+
+import (
+	"context"
+	"fmt"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/mount"
+	"github.com/docker/docker/api/types/network"
+	"github.com/moby/moby/client"
+	"os"
+	"path"
+	"path/filepath"
+)
+
+// CollectorImageName is the default Docker image used
+const CollectorImageName = "ghcr.io/open-telemetry/opentelemetry-collector-releases/opentelemetry-collector-contrib:latest"
+
+const CollectorConfigPath = "/etc/otelcol-contrib/config.yaml"
+
+const sslCertificates = "/etc/ssl/certs/ca-certificates.crt"
+
+type Collector struct {
+	cli         *client.Client
+	containerID string
+}
+
+// NewCollector starts the OpenTelemetry collector container.
+func NewCollector(ctx context.Context, cli *client.Client, net *Networks, params *RunParams, proxy *Proxy) (*Collector, error) {
+	hostCfg := &container.HostConfig{
+		AutoRemove: false,
+	}
+
+	containerCfg := &container.Config{
+		Image: params.CollectorImage,
+		Env: []string{
+			fmt.Sprintf("HTTP_PROXY=%s", proxy.url),
```

Comment:
q: any potential problems if the `url` for `HTTP_PROXY` is `https://` and vice versa for the `HTTPS_PROXY` with a `http://` url

Reply:
`proxy.url` comes without a protocol, so I assume Go handles adding the appropriate protocol as necessary.